apps/cowswap-frontend/src/modules/hooksStore/dapps/PermitHookApp/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import { ButtonPrimary } from '@cowprotocol/ui'
+
+import { useGeneratePermitHook, usePermitInfo, useTokenSupportsPermit } from 'modules/permit'
+
+import { ContentWrapper, Row, Wrapper } from './styled'
+
+import { HookDappProps } from '../../types/hooks'
+import { TradeType } from 'modules/trade'
+import { Token } from '@uniswap/sdk-core'
+import { isAddress } from '@cowprotocol/common-utils'
+
+export function PermitHookApp({ isPreHook, context }: HookDappProps) {
+  const hookToEdit = context.hookToEdit
+  const [tokenAddress, setTokenAddress] = useState<string>()
+  const [spenderAddress, setSpenderAddress] = useState<string>()
+  const generatePermitHook = useGeneratePermitHook()
+  const token = useMemo(() => {
+    try {
+      return new Token(context.chainId, tokenAddress || '', 18)
+    } catch (e) {
+      return null
+    }
+  }, [tokenAddress])
+  const tokenSupportsPermit = useTokenSupportsPermit(token, TradeType.SWAP)
```

Comment:
This hook is a wrapper around `usePermitInfo`, for when the `permitInfo` itself is not needed, only the flag https://github.com/cowprotocol/cowswap/blob/develop/apps/cowswap-frontend/src/modules/permit/hooks/useTokenSupportsPermit.ts
In this case, you do use the `permitInfo`, so `useTokenSupportsPermit` is redundant.